web-app/src/components/HolidayModal/container.js

```diff
@@ -54,49 +56,134 @@ const HolidayModalContainer = Wrapped =>
         })
         .catch(error => {
           swal('Error', `There was an error: ${error.message}. 'error`);
         });
     };
 
-    rejectHoliday = () => {
-      const eventId = this.state.holiday.eventId;
+    assignRejectionResponseText = e =>{
+
+      this.setState({ capturedRejectionReponseText: e.target.value });
+    }
+
+    assignRejectionReasonText = e =>{
 
-      rejectHoliday(eventId)
-        .then(() => {
+      this.setState({ capturedRejectionReasonText: e.target.value });
+    }
+
+    toggleRejectionMessageResponse = rejectionReasonToggleValue =>{
+      
+      this.setState({ rejectionReasonResponse: rejectionReasonToggleValue });
+    
+    }
+
+
+    sendRejectionResponse = () => {
+    
+      const rejectionResponseMessage = this.state.capturedRejectionReponseText;
+      const holidayId = this.state.holiday.holidayId;
+   
+      rejectionResponseMessage( holidayId, rejectionResponseMessage )
+        .then( () => {
+        
           this.setState({
             holiday: {
               ...this.state.holiday,
               eventStatus: {
                 ...this.state.holiday.eventStatus,
                 eventStatusId: holidayStatus.REJECTED,
               },
-            },
+            }, 
+          });
+          Toast({
+            type: 'success',
+            title: 'Holiday Rejected',
+            position: 'top',
+          });
+          
+          this.closeModalResetRejectionReasonView();
+          
+        })
+        .catch(error => {
+          swal('Error', `There was an error: ${error.message}. 'error`);
+        });
+
+
+    }
+
+
+    rejectHoliday = () => {
+      
+      const rejectionReasonText = this.state.capturedRejectionReasonText;
+      const holidayId = this.state.holiday.eventId;
+   
+      rejectHolidayWithMessage( holidayId, rejectionReasonText )
+        .then( () => {
+          this.setState({
+            holiday: {
+              ...this.state.holiday,
+              eventStatus: {
+                ...this.state.holiday.eventStatus,
+                eventStatusId: holidayStatus.REJECTED,
+              },
+            }, 
           });
           Toast({
             type: 'success',
             title: 'Holiday Rejected',
             position: 'top',
           });
+          this.closeModalResetRejectionReasonView();
         })
         .catch(error => {
           swal('Error', `There was an error: ${error.message}. 'error`);
         });
     };
 
+
+    toggleRejectionReasonState = () =>{
+      this.setState({ holidayModalExpansion: false });
+    }
+
+    expandRejectHolidayExplanationText = () =>{
+      this.setState({ holidayModalExpansion: true });
+    }
+
+    closeModalResetRejectionReasonView = () =>{
+      this.setState({ holidayModalExpansion: false, 
+        capturedRejectionReasonText: '', 
+        rejectionReasonResponse: false, 
+        capturedRejectionReponseText: ''  });
+     
+      const { closeModal } = this.props;
```

Comment:
linter would probably pick this up, but best to assign variables at the top of the function